<launch>

    <arg name="simulation" default="false"/>
    <arg name="positioning" default="5"/> <!-- 5 = Visual servoing -->

    <arg name="use_robot_localization" default="true"/>

    <node pkg="dji_sdk" type="dji_sdk_node" name="dji_sdk" output="screen">
        <!-- node parameters -->
        <param name="serial_name" type="string" value="/dev/ttyUSB0"/>
        <param name="baud_rate" type="int" value="230400"/>
        <param name="app_id" type="int" value="1082161"/>
        <param name="app_version" type="int" value="1"/>
        <param name="align_time" type="bool" value="false"/>
        <param name="enc_key" type="string" value="aa7f4935a5c220dd2d7e1ca4f0fcc75d0b7e51b99aad7d44f74e7c468c639e8c"/>
        <param name="use_broadcast" type="bool" value="false"/>
    </node>

    <include file="$(find dtu_controller)/launch/arducam.launch" />

    <node pkg="visual_tracker_w_gui" type="csrt_tracker.py" name="csrt_tracker" output="screen" />

    <node pkg="rosserial_python" type="serial_node.py" name="rosserial_node" args="/dev/ttyACM0" output="screen" />

    <!-- OLD NODE USE LAUNCH INSTEAD-->
    <!-- <node pkg="dtu_controller" type="ransac_laser_w_subscriber.py" name="ransac_laser" output="screen" /> -->
    <include file="$(find dtu_controller)/launch/ransac_laser.launch" />
    
    <group if="$(arg use_robot_localization)">
        <include file="$(find dtu_controller)/launch/robot_localization.launch" />
        <node pkg="dtu_controller" type="odo_2_twist.py" name="odorepub" output="screen" />
    </group>

    <group ns="dtu_controller">
        <node pkg="dtu_controller" type="robot_localization_rpy_position" name="robot_localization_rpy_position" output="screen">
            <rosparam command="load" file="$(find dtu_controller)/parameters/x_y_zvel_yawrate_pids_winter.yaml" />
        </node>

        <group if="$(eval arg('use_robot_localization') == 0 )">
            <node pkg="dtu_controller" type="position_observer_node" name="position_observer" output="screen">
                <rosparam command="load" file="$(find dtu_controller)/parameters/observer_params.yaml" />
                <param name="simulation" type="bool" value="$(arg simulation)"/>
                <param name="positioning" type="int" value="$(arg positioning)"/>
            </node>
        </group>

        <node pkg="dtu_controller" type="autonomous_wall_touch" name="autonomous_wall_touch" output="screen" />
    </group>



</launch>
